### DISCUSSION TITLE
Gemini AI Extention

### DISCUSSION THREAD

**Discussion by aligundogar** [FreshRSS/FreshRSS#6434] [Ideas]
I have app on docker container with reverse proxy. Im thinking of gemini api integration with it. How can I do that? I saw ext. News-Assistant but its based on Openai.

**Comment by aligundogar** [comment]
Thinking of run it with FreshRSS, is there anyone to can help me?

const Parser = require('rss-parser');
const {
  GoogleGenerativeAI,
  HarmCategory,
  HarmBlockThreshold,
} = require('@google/generative-ai');

const MODEL_NAME = 'gemini-1.5-pro-latest';
const API_KEY = 'XXX';

const parser = new Parser();
const feedURL = 'https://test.com/feed/';

async function getRSSData() {
  try {
    const feed = await parser.parseURL(feedURL);
    const latestItem = feed.items[0]; 

  
    const rssContent = `${latestItem.title} ${latestItem.contentSnippet}`;

    return rssContent;
  } catch (err) {
    console.error('RSS verisi alınamadı:', err);
    return null;
  }
}

async function runChat() {
  const rssContent = await getRSSData();

  if (!rssContent) {
    console.log('RSS verisi alınamadı.');
    return;
  }

  const genAI = new GoogleGenerativeAI(API_KEY);
  const model = genAI.getGenerativeModel({ model: MODEL_NAME });

  const generationConfig = {
    temperature: 1,
    topK: 0,
    topP: 0.95,
    maxOutputTokens: 8192,
  };

  const safetySettings = [
    {
      category: HarmCategory.HARM_CATEGORY_HARASSMENT,
      threshold: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
    },
    {
      category: HarmCategory.HARM_CATEGORY_HATE_SPEECH,
      threshold: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
    },
    {
      category: HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT,
      threshold: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
    },
    {
      category: HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT,
      threshold: HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE,
    },
  ];

  const chat = model.startChat({
    generationConfig,
    safetySettings,
    history: [],
  });

  const result = await chat.sendMessage(rssContent);

  const response = result.response;
  console.log(response.text());
}

runChat();

**Comment by Frenzie** [comment]
Integration to do what exactly?

**Reply by aligundogar** [reply]
Use freshrss feeds and tunned gemini LLM together. I wanted to specialize bot's ansews for analyzing news via documents, pdfs wich I uploaded. I created bot with edited behavior and documents.

**Reply by Frenzie** [reply]
Under settings → profile you can find an authentication token to access the feed without authentication.

**Reply by aligundogar** [reply]
thank you

**Comment by Alkarex** [comment]
Related: AI classification implemented in:
* https://github.com/FreshRSS/Extensions/pull/458
